### ISSUE TITLE
[wptserve] Figure out which string type URLs should be

### ISSUE THREAD

**Issue by Hexcles** [web-platform-tests/wpt#23362]
This is probably one of the last few properties on `Request` that are still using native strings and it is widely used.

We probably want to use binary strings / bytes for this.

**Comment by foolip** [comment]
@Hexcles this came up in the infra issue triage. What's the remaining work here, and is anyone assigned to do it?

**Comment by stephenmcgruer** [comment]
I don't think this is roadmap at this time. This is code-health cleanup work, so dropping to backlog.
